agent/cache/cache.go

```diff
@@ -121,23 +124,26 @@ type ResultMeta struct {
 
 // Options are options for the Cache.
 type Options struct {
-	// Nothing currently, reserved.
+	// EntryFetchMaxBurst max burst size of RateLimit for a single cache entry
+	EntryFetchMaxBurst int
```

Comment:
According to https://godoc.org/golang.org/x/time/rate#Limiter.Burst it's the number of tokens per `Wait` call, which I believe is also going to be 1. Do we need to make this configurable? Or can we remove it and always use 1 ?

Reply:
Yes, the bust rate is important: it allows the rate-limit to be overridden for burst-size time (defaults to 2), but globally respect the rate-limit (in our case, we will probably configure it with a value around 4 or 5 I think and 0.1 qps => you can have quick updates on a stable service, but when too many updates are firing on a service, you get rate-limited at 1 query every 10s) I use the same configuration as the rate-limit you already implemented for RPC which includes burst size as well.

Reply:
I think either I have misunderstood the docs I linked, or the docs are misleading, or it doesn't work quite that way. The docs suggest this only impacts `Wait()`, and we always use `Wait()` with a value of 1 , because we only ever fetch an entry one at a time. I'll need to read the code to better understand how burst works.

Reply:
@dnephin This is exactly how it works for RPC Max Rate limit as well: https://github.com/hashicorp/consul/blob/master/agent/agent.go#L1404

Reply:
Thank you both for the clarification!

It sounds like this setting is actually the bucket size. I wonder if hat would have been a better name for the config field. I guess since we already call this burst elsewhere in the config we probably have to keep it burst here as well.

Reply:
To me, burst make sense from a functional point of view: the operator want to allow burst of data until the rate limit is reached. Bucket is more technical about the implementation. But as you highlighted it, it is anyway already used that way in RPC, so I think it make sense to change it